src/libasr/codegen/asr_to_x86.cpp

```diff
@@ -523,8 +523,7 @@ class ASRToX86Visitor : public ASR::BaseVisitor<ASRToX86Visitor>
     }
 
     void visit_SubroutineCall(const ASR::SubroutineCall_t &x) {
-        ASR::Function_t *s = ASR::down_cast<ASR::Function_t>(
-            ASRUtils::symbol_get_past_external(x.m_name));
+        ASR::Function_t *s = ASR::down_cast<ASR::Function_t>(x.m_name);
```

Comment:
I think this needs to be reverted (https://github.com/lcompilers/lpython/commit/39aed2691c7128d125af0332bfd895d8510d7314).